homeassistant/components/remote/harmony.py

```diff
@@ -0,0 +1,163 @@
+        self._current_activity = None
+        self._default_activity = activity
+        self._token = token
+        self._config_path = path
+        _LOGGER.debug('retrieving harmony config using token: ' + token)
+        self._config = pyharmony.ha_get_config(self.token, host, port)
+        if not Path(self._config_path).is_file():
+            _LOGGER.debug('writing harmony configuration to file: ' + path)
+            pyharmony.ha_write_config_file(self._config, self._config_path)
+
+    @property
+    def name(self):
+        """Return the Harmony device's name."""
+        return self._name
+
+    @property
+    def token(self):
+        """Return the token of the Harmony device."""
+        return self._token
+
+    @property
+    def config(self):
+        """Return the configuration of the Harmony device."""
+        return self._config
+
+    @property
+    def device_state_attributes(self):
+        """Add platform specific attributes."""
+        return {'current_activity': self._current_activity}
+
+    @property
+    def is_on(self):
+        """Return False if PowerOff is the current activity, otherwise True."""
+        return self._current_activity != 'PowerOff'
+
+    def update(self):
+        """Return current activity."""
+        import pyharmony
+        name = self._name
+        _LOGGER.debug('polling ' + name + ' for current activity')
+        state = pyharmony.ha_get_current_activity(self._token,
+                                                  self._config,
+                                                  self._ip,
+                                                  self._port)
+        _LOGGER.debug(name + '\'s current activity reported as: ' + state)
+        self._current_activity = state
+        self._state = bool(state != 'PowerOff')
+
+    def turn_on(self, **kwargs):
+        """Start an activity from the Harmony device."""
+        import pyharmony
+        if kwargs[ATTR_ACTIVITY]:
+            activity = kwargs[ATTR_ACTIVITY]
+        else:
+            activity = self._default_activity
+
+        if activity != 'Not_Set':
+            pyharmony.ha_start_activity(self._token,
+                                        self._ip,
+                                        self._port,
+                                        self._config,
+                                        activity)
+            self._state = True
+        else:
+            _LOGGER.error('No activity specified with turn_on service')
+
+    def turn_off(self):
+        """Start the PowerOff activity."""
+        import pyharmony
+        pyharmony.ha_power_off(self._token, self._ip, self._port)
+
+    def send_command(self, **kwargs):
+        """Send a command to one device."""
+        import pyharmony
+        pyharmony.ha_send_command(self._token, self._ip,
+                                  self._port, kwargs[ATTR_DEVICE],
+                                  kwargs[ATTR_COMMAND])
+
+    def sync(self):

```

Comment:
This seems like a very specific use case. Remove sync as a Remote level service and add a harmony specific sync service (`harmony_sync`) that gets added by this platform.

See Sonos for an example of registering services inside your platform. Name should be prefixed with platform name.